### PR TITLE
MOD-7320: Support ubuntu noble numbat

### PR DESCRIPTION
**Pull request by kei-nan**
Add support for ubuntu noble numbat for amd64 and arm64

A clear and concise description of what the PR is solving, including:
1. Current State: Github workflows do not support ubuntu noble numbat
2. Change: 
    - Add support for ubuntu noble numbat to the workflows
    - If RedisJSON module is not found for the OS/arch, it is build from source.
4. Outcome: Ubuntu Noble Numbat should be added to our github workflow

**Which issues this PR fixes**
1. MOD-7320

**Main objects this PR modified**
1. Github workflows

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: .github/workflows/flow-build-artifacts.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ on:
         options:
           - all
           - ubuntu:jammy
+          - ubuntu:noble
```
Comment (raz-mon) [inline .github/workflows/flow-build-artifacts.yml:11]: For good order's sake, switch them :)

File: sbin/get-redisjson [hunk 2]
```diff
@@ -58,6 +58,8 @@ else
 	[[ $arch == arm64v8 ]] && arch=aarch64
 
 	if [[ $dist == ubuntu ]]; then
+	  if [[ $nick == noble ]]; then
+    	nick="ubuntu24.04"
```
Comment (raz-mon) [inline sbin/get-redisjson:62]: Use the same format as below (and below `if` --> `elif`)

File: sbin/get-redisjson [hunk 3]
```diff
@@ -149,16 +151,20 @@ WORK_DIR=$(mktemp -d /tmp/${MOD_NAME}.XXXXXX)
...
 	F_MOD_RAMP="${WORK_DIR}/${MOD_RAMP}"
 	if [[ $FORCE == 1 || ! -f $F_MOD_RAMP ]]; then
-		echo "Download ${MOD_NAME} [${MOD_S3_URL}/${MOD_RAMP}] ..."
-		runn wget -P ${WORK_DIR} ${MOD_S3_URL}/${MOD_RAMP}
-		if [[ $NOP != 1 && $? != 0 ]]; then
-			if [[ BUILD == 0 ]]; then
-				eprint "Cannot download binary artifacts - build ${MOD_NAME} and set REPO_PATH"
-				exit 1
+		# check if redisjson module exists in the s3 bucket
+		if wget -q --spider ${WORK_DIR} ${MOD_S3_URL}/${MOD_RAMP}; then
```
Comment (GuyAv46) [inline sbin/get-redisjson:155]: so it's just a prettier validation, not a solution in case it is missing right?

Comment (nafraf) [reply sbin/get-redisjson]: The condition was wrong, but now it is fixed.
If the module exists, it is downloaded, if the module is not found, it is build from source code.

**Review by GuyAv46 — APPROVED** [review]

**Pull request merged**